File: 1231-replace-elements-with-greatest-element-on-right-side/solution.go
func replaceElements(arr []int) []int {
    n := len(arr)

    maxEl := -1
    for i := n - 1; i >= 0; i-- {
        temp := arr[i]
        arr[i] = maxEl

        if temp > maxEl {
            maxEl = temp
        }
    }
    
    return arr
}

File: 1231-replace-elements-with-greatest-element-on-right-side/solution.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        n = len(arr)

        curr_max = arr[n - 1]
        for i in range(n - 1, -1, -1):
            temp = arr[i]
            arr[i] = curr_max
            if temp > curr_max:
                curr_max = temp

        arr[n - 1] = -1
        return arr
